_studio/mfx_lib/encode_hw/mpeg2/src/mfx_mpeg2_encode_utils_hw.cpp

```diff
@@ -948,6 +948,8 @@ namespace MPEG2EncoderHW
         CHECK_CODEC_ID(par->mfx.CodecId, MFX_CODEC_MPEG2);
         MFX_CHECK (CheckExtendedBuffers(par) == MFX_ERR_NONE, MFX_ERR_INVALID_VIDEO_PARAM);
 
+        mfxStatus sts = core->IsGuidSupported(DXVA2_Intel_Encode_MPEG2, par);
```

Comment:
Will this function's call work for Linux as well?

Reply:
Yes. Mpeg2 support FullHD and IsGuidSupported will check this resolution. Check resolution will not be needed on 8K in 990 and 984 strings. As well as IsGuidSupported checks encode support on platform.